website/addons/mendeley/model.py

```diff
@@ -86,43 +88,77 @@ def citation_lists(self, extract_folder):
         ]
         return [all_documents] + serialized_folders
 
-    def get_list(self, list_id='ROOT'):
+    def _marker_from_page(self, page):
+        if page is None:
+            return None
+        next_link = page.rsp.links.get('next')
+        if next_link is None:
+            return None
+        link = next_link['url']
+        qs = link.split('?')[-1]
+        return parse_qs(qs).get('marker')
+
+    def get_list(self, list_id='ROOT', next_page=None):
         """Get a single CitationList
         :param str list_id: ID for a Mendeley folder. Optional.
         :return CitationList: CitationList for the folder, or for all documents
         """
-        if list_id == 'ROOT':
+        if list_id == 'ROOT' or list_id is None:
             folder = None
         else:
             folder = self.client.folders.get(list_id)
 
+        citations = None
+        page = None
         if folder:
-            return self._citations_for_mendeley_folder(folder)
-        return self._citations_for_mendeley_user()
+            citations, page = self._citations_for_mendeley_folder(folder, next_page)
+        else:
+            citations, page = self._citations_for_mendeley_user(next_page)
+        marker = self._marker_from_page(page)
+        return citations, marker
 
     def _folder_metadata(self, folder_id):
         folder = self.client.folders.get(folder_id)
         return folder
 
-    def _citations_for_mendeley_folder(self, folder):
+    def _citations_for_mendeley_folder(self, folder, next_page):
+        documents = None
+        if hasattr(next_page, 'next_page'):
+            documents = next_page.next_page
+        elif next_page is None:
+            documents = folder.documents.list(page_size=100)
+        else:
+            documents = folder.documents.list(page_size=100, marker=next_page)
 
-        document_ids = [
+        document_ids = set([
             document.id
-            for document in folder.documents.iter(page_size=500)
-        ]
-        citations = {
-            citation['id']: citation
-            for citation in self._citations_for_mendeley_user()
-        }
-        return map(lambda id: citations[id], document_ids)
-
-    def _citations_for_mendeley_user(self):
+            for document in documents.items
+        ])
+        citations = {}
+        next_page = None
+        while len(document_ids - set((citations or {}).keys())) > 0:
```

Comment:
The Mendeley API returns folder contents as a list of ids. We need to fetch citations until all of the citations ids in a given folder are retrieved.